pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -62,41 +68,97 @@ func init() {
 }
 
 type CloudWatchService struct {
-	LogsService *LogsService `inject:""`
+	LogsService          *LogsService          `inject:""`
+	BackendPluginManager backendplugin.Manager `inject:""`
 }
 
 func (s *CloudWatchService) Init() error {
 	plog.Debug("initing")
 
-	tsdb.RegisterTsdbQueryEndpoint("cloudwatch", func(ds *models.DataSource) (tsdb.TsdbQueryEndpoint, error) {
-		return newExecutor(s.LogsService), nil
+	im := datasource.NewInstanceManager(NewInstanceSettings())
+
+	factory := coreplugin.New(backend.ServeOpts{
+		QueryDataHandler: newExecutor(s.LogsService, im),
 	})
 
+	if err := s.BackendPluginManager.Register("cloudwatch", factory); err != nil {
+		plog.Error("Failed to register plugin", "error", err)
+	}
 	return nil
 }
 
-func newExecutor(logsService *LogsService) *cloudWatchExecutor {
+func newExecutor(logsService *LogsService, im instancemgmt.InstanceManager) *cloudWatchExecutor {
 	return &cloudWatchExecutor{
 		logsService: logsService,
+		im:          im,
+	}
+}
+
+func NewInstanceSettings() datasource.InstanceFactoryFunc {
+	return func(settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+		var jsonData map[string]string
+
+		err := json.Unmarshal(settings.JSONData, &jsonData)
```

Comment:
This is something I haven't really thought a lot of before. We would marshal the JSON here https://github.com/grafana/grafana/blob/a18e4d8d6437a0cf57e8ddebe61d36d5ce081e25/pkg/plugins/backendplugin/coreplugin/query_endpoint_adapter.go#L29

And then here we Unmarshal. Might need to think about how we can skip this. Maybe the answer is just to migrate away from tsdb.QueryEndpoint/models and we wouldn't need all of these wrappers/adapters from/to SDK/tsdb things. But definitely something to take into consideration what affect this might have in regards to performance and such during the migration path of things.